inst/stan/functions/combine_effects.stan

```diff
@@ -47,18 +47,13 @@ vector combine_effects(array[] real intercept, vector beta, matrix design,
                        vector beta_sd, matrix sd_design, int add_intercept) {
   int nobs = rows(design);
   int neffs = num_elements(beta);
-  int sds = num_elements(beta_sd);
-  vector[neffs + add_intercept] scaled_beta;
-  vector[sds + 1] ext_beta_sd;
-  ext_beta_sd[1] =  1.0;
   if (neffs) {
-    ext_beta_sd[2:(sds+1)] = beta_sd;
+    vector[num_elements(beta_sd) + 1] ext_beta_sd = append_row(1.0, beta_sd);
```

Comment:
This comment blocks accidental merging. Leave unresolved.

Reply:
only allocate this if actually being used - seems reasonable. no idea if the append_row vs slice assignment is faster.

Reply:
this might be mildy clearer as `vector[1 + num_elements(beta_sd)] ext_beta_sd = ...`, given what's happening on the RHS

Reply:
Agree - updated